### ISSUE TITLE
bug: no glibc available info now

### ISSUE THREAD

**Issue by mckaygerhard** [lpereira/hardinfo#79]
i cannot track from where commit the libc6 info cannot be tracket.. but now its unkown, i tested in Debian jeesie, Devuan jeesie, Debian wheeze, and only works in Debian squeeze and lenny works too perfectly, see picture:

![errorcglibchardinfo](https://user-images.githubusercontent.com/1571724/27542667-f35d976a-5a55-11e7-8917-43efad1e1938.png)

Debian in recent packages has a patch but its unusable (was using ldconfig -V and strip the version info) due neither works.

**Comment by mckaygerhard** [comment]
tested in linux x86 embebed (daruma) and works, its equivalent with debian squeeze, due recent stable versions are too slower and vendor does not want to update..

**Comment by mckaygerhard** [comment]
seems the problem its related only to Debian, due in alpine works..

**Comment by mckaygerhard** [comment]
ok, the top patch https://developer.gnome.org/glib/stable/glib-Spawning-Processes.html#g-spawn-command-line-async seem both calls are glib older compatible.. i'll test in other later in night and report feedback

**Comment by mckaygerhard** [comment]
tested in various hardwares with debian squeeze, worked:

**debian older << 7 , squeeze, works:** tested in dell r15, dell powered, dell t110 and asus eepc with debian squeeze 6.0, 
**debian newers >> 7 (jessie, sid), dont:** tested in debian jeesie not working, see image, ill test now in devuan not working neither.. i do not tested in other distros rather debian derived!

see images:

![hardinfo-error-gettingglibcerror](https://user-images.githubusercontent.com/1571724/28135096-1535651a-6713-11e7-821d-b159b9d46e37.png)

![hardinfo-glibc6-debiansqueeze](https://user-images.githubusercontent.com/1571724/28135106-1fd3776e-6713-11e7-8e42-c0dec212892c.png)

**Comment by bp0** [comment]
The format string changed in the fix. It no longer has the (stable) part, so you're testing a version of hardinfo that doesn't have the recent fixes to libc detection.

**Comment by mckaygerhard** [comment]
hi @bp0 i tested with your brand, and also with the last brand of stable at yesterday.. and now i tested today with the lasted in git.. and same results..

i not tested in debian wheeze yet.. i think its a problem in the way of glibc returns the version strings.. i not have another distro to test .. but i could install today a winbuntu and test.. , i'll make it with kubuntu-trinity and report feedback...

**Comment by mckaygerhard** [comment]
tested with recent patches : i think its something with debian.. i'll test with other distro in virtualbox today at night and report feedback

![2017-07-20-105912_1366x768_scrot](https://user-images.githubusercontent.com/1571724/28424009-90c6712e-6d3a-11e7-8a73-30d45743872c.png)

**Comment by bp0** [comment]
What is your output of `ldconfig -V` ?

**Comment by mckaygerhard** [comment]
in newer debians its:
``` bash
root@syslaptop1:/home/systemas/dwhelper# ldconfig -V
ldconfig (Debian GLIBC 2.19-18+deb8u7) 2.19
Copyright (C) 2014 Free Software Foundation, Inc.
Esto es software libre; vea el código fuente para las condiciones de copia.
No hay NINGUNA garantía; ni siquiera de COMERCIABILIDAD o IDONEIDAD PARA UN
FIN DETERMINADO.
Escrito por Andreas Jaeger.
root@syslaptop1:/home/systemas/dwhelper# 
```
in olders (where hardinfo recent commits are working):
``` bash
root@vnxemil:~# ldconfig -V
ldconfig (Debian EGLIBC 2.11.3-3) 2.11.3
Copyright (C) 2009 Free Software Foundation, Inc.
Esto es software libre; vea el código fuente para las condiciones de copia.
No hay NINGUNA garantía; ni siquiera de COMERCIABILIDAD o IDONEIDAD PARA UN
FIN DETERMINADO.
Escrito por Andreas Jaeger.
root@vnxemil:~#
```

the "EGLIBC" are due squeeze change from GLIB to EGLIB and then return to GLIB but in lennt the output it still : 

``` bash
sysremaster:~# ldconfig -V
ldconfig (GNU libc) 2.7
Copyright (C) 2007 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
Written by Andreas Jaeger.
sysremaster:~# 
```

in olders debians the libc6 path are:
`/lib/libc/libc-2.7.so`
in newer are same:
`/lib/libc/libc-2.11.3.so`

**Comment by bp0** [comment]
Code:
https://github.com/lpereira/hardinfo/blob/master/modules/computer/os.c#L25

* All match either "GLIBC" or "GNU libc"
* All have ldconfig as skipped first word

So what is left? Is the output in stdout or stderr?

**Comment by mckaygerhard** [comment]
the comparison are case-sensitive or in-sensitive?

**Comment by mckaygerhard** [comment]
fixed: closed with 620997880825682ac43eec3e0fcc9c40407dd5c4  image:

![workingglibcdetect](https://user-images.githubusercontent.com/1571724/28465156-c4f2cff4-6df5-11e7-8ff3-fede2098356f.png)
